source/extensions/filters/common/ratelimit/ratelimit.h

```diff
@@ -92,6 +92,11 @@ class Client {
                      const std::vector<Envoy::RateLimit::Descriptor>& descriptors,
                      Tracing::Span& parent_span, OptRef<const StreamInfo::StreamInfo> stream_info,
                      uint32_t hits_addend) PURE;
+
+  /**
+   * Returns streamInfo of the current request if possible. By default just return a nullptr.
+   */
+  virtual StreamInfo::StreamInfo const* streamInfo() const { return nullptr; }
```

Comment:
The client in the future will be refactored to be shared by multiple requests, I actually don't think this is a correct way to expose the inner stream info.